File: src/main/java/pl/arturczopek/mycoach/repository/PriceRepository.java
package pl.arturczopek.mycoach.repository;

import org.springframework.data.repository.CrudRepository;
import pl.arturczopek.mycoach.model.database.Price;

import java.util.List;

/**
 * @Author Artur Czopek
 * @Date 09-10-2016
 */

public interface PriceRepository extends CrudRepository<Price, Long> {

    List<Price> findByProductIdOrderByPriceDateAsc(long productId);

    void deleteByProductIdOrderByPriceDate(long productId);
}

File: src/main/kotlin/pl/arturczopek/mycoach/repository/PriceRepository.kt
package pl.arturczopek.mycoach.repository

import org.springframework.data.repository.CrudRepository
import pl.arturczopek.mycoach.model.database.Price

/**
 * @Author Artur Czopek
 * @Date 09-10-2016
 */

interface PriceRepository : CrudRepository<Price, Long> {

    fun findByProductIdOrderByPriceDateAsc(productId: Long): List<Price>

    fun deleteByProductIdOrderByPriceDate(productId: Long)
}
